### ISSUE TITLE
Fix S6544 (`no-misused-promises`): Do not flag not awaited, resolved or error-handled promises

### ISSUE THREAD

**Issue by ilia-kebets-sonarsource** [SonarSource/SonarJS#3891]
We don't want to flag the following use cases:
```javascript
async function foo() {
	try {
		await bar()
	} catch (err) {
		// handle error
	}
}
foo(); // <- Noncompliant
```

Therefore, we should remove the `no-floating-promises` rule from the implementation.

**Comment by yassin-kammoun-sonarsource** [comment]
This seems rather a drastic decision to remove completely the support of a rule just because a particular code pattern is noisy. Couldn't we just improve the rule? We could maybe filter out false positives or reimplement the rule to cover simpler cases, couldn't we?
